### PR TITLE
Refactor and fix Video manager issues #10348

### PR DESCRIPTION
**Pull request by zdanek**
This fixes issues covered in Bug report #10348

That is 
* playing video even though source is switched to `Video stream disabled`
* displaying video preview window even though source is switched to `Video stream disabled`
* utilizing UVC and GStreamer even though source is switched to `Video stream disabled`

### CODE REVIEW

**Comment by WTPENGUIN** [comment]
".ts" files are included in the commit. Is there any other reason?

And there is UVC Video Loader in FlyViewVideo.qml. It looks good to fix this too.

```QML
//-- UVC Video (USB Camera or Video Device)
    Loader {
        id:             cameraLoader
        anchors.fill:   parent
        visible:        !QGroundControl.videoManager.isGStreamer
        source:         QGroundControl.videoManager.uvcEnabled ? "qrc:/qml/FlightDisplayViewUVC.qml" : "qrc:/qml/FlightDisplayViewDummy.qml"
    }
```

to

```QML
//-- UVC Video (USB Camera or Video Device)
    Loader {
        id:             cameraLoader
        anchors.fill:   parent
        visible:        QGroundControl.videoManager.isUvc
        source:         QGroundControl.videoManager.uvcEnabled ? "qrc:/qml/FlightDisplayViewUVC.qml" : "qrc:/qml/FlightDisplayViewDummy.qml"
    }
```

**Comment by zdanek** [comment]
@WTPENGUIN .ts files are because I've changed "File Format" label to "Record File Format". I thought that "File Format" is a key to all .ts translations. Isn't it? If I change the key, all .ts files should be fixed. Am I wrong?

Regarding fix in FlyViewVideo.qml - thank you for pointing out. I will check it.

Currently I'm having some errors in Android build. I've missed it on my PC. I have to check it but suddenly Android version is not building in my Qt Creator. Even master version so I suppose it's not my fault. I have to fix it and then check Android error in my PR. I will post fixes ASAP.

@WTPENGUIN can you respond to my reasoning regarding .ts files?

**Comment by WTPENGUIN** [comment]
@zdanek

I looked in the "translations folder" of the source code. I made a 'guess' that there might be an automated system. To be precise, I think the maintainer should answer.

**Comment by zdanek** [comment]
@DonLakeFlyer are you able to check my PR?

**Comment by zdanek** [comment]
Hey @bkueng  can you check my PR?

**Comment by DonLakeFlyer** [comment]
Sorry I haven't been very involved with the video stuff.

**Comment by zdanek** [comment]
Who can help?

pon., 22 sie 2022, 18:57 użytkownik Don Gagne ***@***.***>
napisał:

**Comment by DonLakeFlyer** [comment]
The Blue Robotics folks are the one who know the most about video.

**Comment by zdanek** [comment]
@Williangalvani @patrickelectric review please. Video is a real mess in QGC and there are other issues with it - for example RTSP via gstreamer does not work on Android. I'd like to fix it all but first I'd like you to check my PR. Any feedback would be appreciated. Let's get on with this code :) 
Thank you :heart:

File: src/FlightDisplay/FlightDisplayViewUVC.qml
Comment (patrickelectric) [inline src/FlightDisplay/FlightDisplayViewUVC.qml:57]: The indentation is wrong

File: src/VideoManager/VideoManager.cc
Comment (patrickelectric) [inline src/VideoManager/VideoManager.cc:113]: That is happening on:
https://github.com/mavlink/qgroundcontrol/pull/10349/files#diff-1e8e0d1887083c7f6f0dd38e78042c25afb687cc60b955bf572f039ed9547fd0R493

**Review by patrickelectric — CHANGES_REQUESTED** [review]
The video stream is not working here on both APK and also native build:
image

File: src/VideoManager/VideoManager.cc
Comment (zdanek) [reply src/VideoManager/VideoManager.cc:113]: @patrickelectric 
> The video stream is not working here on both APK and also native build: 

1) it should work on native build - what OS do you have? 
2) it might not work on Android since a) I did not test it - sorry for that - **I will test it** b) I rebased to master and there gstreamer does not work as I mentioned in conversation with @fredowski - **I will check it and fix** prior to this PR so we have clean resolution before my tidy works in current PR

Thank you for rest of the feedback. I will fix it.

What about notices by @WTPENGUIN Shall I revert changes in `ts` files? I don't know what are translation mechanisms and maybe he's right and I shall not touch `ts` files - which would be strange though.

**Review by patrickelectric — COMMENTED** [review]
>> The video stream is not working here on both APK and also native build:

> it should work on native build - what OS do you have?
> it might not work on Android since a) I did not test it - sorry for that - I will test it b) I rebased to master and there gstreamer does not work as I mentioned in conversation with @fredowski - I will check it and fix prior to this PR so we have clean resolution before my tidy works in current PR
> Thank you for rest of the feedback. I will fix it.
> 
> What about notices by @WTPENGUIN Shall I revert changes in ts files? I don't know what are translation mechanisms and maybe he's right and I shall not touch ts files - which would be strange though.

I did test on my Linux Arch, Android and Mac M1, all of them show no video using the artifacts from this PR.
To test, you can just run:

`gst-launch-1.0 videotestsrc ! video/x-raw,width=640,height=480 ! videoconvert ! x264enc ! rtph264pay ! udpsink host=127.0.0.1 port=5600`

You can touch the `.ts` files, what @WTPENGUIN is suggesting is that the changes are unrelated to the video rework, you can improve that using a different commit or creating a new PR, since we can merge it easily compared to the code changes. 
For more information about commit organization and others good practices:
https://patrickelectric.work/tags/how-to-rock/

**Comment by fredowski** [comment]
@zdanek: During my experiments I noticed the bugs that you plan to fix... When I look at your changes I notice the following:

  * The files in the translations directory are generated. As @patrickelectric mentioned, you can maybe keep those out of this PR. In other projects the translators receive an update only once a substantial number of changes have been done or you are closer to a release.
  * The CI pipeline is now working for Android again. Maybe you can remove your preliminary fix and rebase to master? 
  * There are some changes for MacOS and all platforms regarding RTSP and recording in PR #10396. Once that is merged the video related problems can be checked also on MacOS. But that is probably no prerequisite.

**Comment by fredowski** [comment]
I guess issue #5608 is related to this PR?

**Comment by zdanek** [comment]
Yes!

czw., 8 wrz 2022, 22:14 użytkownik Friedrich Beckmann <
***@***.***> napisał:

File: src/FlightDisplay/FlightDisplayViewUVC.qml
Comment (zdanek) [reply src/FlightDisplay/FlightDisplayViewUVC.qml:57]: @patrickelectric identation is good because I've pushed `onVisibleChanged` into `VideoOutput` element since this event is a part of `VideoOutput`.
Please unfold lines 40-50 to see it.

File: src/VideoManager/VideoManager.cc
Comment (zdanek) [reply src/VideoManager/VideoManager.cc:113]: fixed

**Comment by zdanek** [comment]
@patrickelectric will look at ma latest changes? I'm sure it's ready be reviewed.

**Comment by mrpollo** [comment]
Looks like we should be testing this again with the updates to master branch, can you please give it a try? @patrickelectric @bkueng ?

**Comment by zdanek** [comment]
@mrpollo @patrickelectric @bkueng can you please review and push this topic further. This is a bit frustrating to wait months for a code to be reviewed while you're pushing your own changes day by day.

**Comment by mrpollo** [comment]
IMO this is ready to go in, but I can't test this, let me see if @patrickelectric or someone else from the community can help do a quick sanity check

**Comment by farhangnaderi** [comment]
@mrpollo it seems ok for me:
image

**Comment by mrpollo** [comment]
Thanks @farhangnaderi it looks like this is ready to be merged.

**Comment by mrpollo** [comment]
Thanks for the contribution @zdanek 🎉

**Pull request merged**
